mongoose_factory.js

```diff
@@ -1,6 +1,9 @@
 const mongoose = require('mongoose');
 
-mongoose.connect(process.env.MONGO_URI, {useNewUrlParser: true, useUnifiedTopology: true});
+mongoose.connect(
+    `mongodb://${process.env.MONGO_USERNAME}:${process.env.MONGO_PASSWORD}@${process.env.MONGO_HOST}:${process.env.MONGO_PORT}/${process.env.MONGO_DB}`, 
```

Comment:
One more thing to work with Mongo Atlas. It uses `mongodb+srv` to flag it as a DNS entry that it has to lookup for mongod.

In this case, there is also no port for mongo.

To solve for this, I think you can just add a flag in the .env for using srv i.e. `MONGO_USE_SRV`, which will automatically exclude the port from the string and use `mongodb+srv://` for the beginning instead.

Reply:
https://www.mongodb.com/docs/manual/reference/connection-string/#dns-seed-list-connection-format